### PR TITLE
chore(ssa refactor): Add method to handle black box functions in Acir gen

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

This will handle calls to black box functions

## Problem\*

Resolves 

## Summary\*

This PR sets out to

### Example

Before:

```

```

After:

```

```

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by kevaundray** [comment]
Still a WIP while :

- I ponder over the API absraction we want to use for this. (Left some TODOs to specify which parts I plan to resolve) 
- It handles blackbox functions, but we also have builtins which are not handled.
- An example calling intrinsics and builtins are missing
- Clean up TODOs

File: crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs [hunk 1]
```diff
@@ -331,6 +336,218 @@ impl AcirContext {
...
+                    // by doing the following:
+                    //
+                    // ```
+                    // call_intrinsic(x as u8)
+                    // ```
+                    //
+                    // The `as u8` specifies that we want to take 8 bits from the `x`
+                    // variable.
+                    //
+                    // TODO(Jake): There were discussions about the SSA IR optimizing out range
```
Comment (kevaundray) [inline crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs:466]: @jfecher pinging for your insight here

File: crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs [hunk 2]
```diff
@@ -331,6 +336,218 @@ impl AcirContext {
...
+            // Inputs for fixed based scalar multiplication
+            // is just a scalar
+            BlackBoxFunc::FixedBaseScalarMul => Some(1),
+        }
+    }
+    /// This function will return the number of outputs that a blackbox function
+    /// expects. Returning `None` if there is no expectation.
+    ///
+    /// TODO: With recursion, this becomes quite annoying, since that
+    /// TODO: is backend specific
```
Comment (kevaundray) [inline crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs:524]: We don't have recursion yet, so removing this until we get recursion is okay

**Review by joss-aztec — APPROVED** [review]

File: crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs (hunk 1 above)
Comment (jfecher) [reply crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs:398]: We could consider removing the actual cast instruction and reinterpreting the type of x without inserting an extra, unnecessary range constraint. So I think there are ways to keep this optimization and work around this problem, but it is also not the most important optimization so it is just something to keep in mind for the future.

**Review by jfecher — APPROVED** [review]

File: crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs (hunk 1 above)
Comment (kevaundray) [reply crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs]: Created issue #1467 to track this

**Review by joss-aztec — APPROVED** [review]

**Pull request merged**